subprojects/core/src/main/java/org/gradle/caching/configuration/internal/DefaultBuildCacheConfiguration.java

```diff
@@ -38,47 +37,39 @@
 
     private final Instantiator instantiator;
 
-    private BuildCache local;
+    private final DirectoryBuildCache local;
     private BuildCache remote;
 
     private final Set<BuildCacheServiceRegistration> registrations;
 
     public DefaultBuildCacheConfiguration(Instantiator instantiator, List<BuildCacheServiceRegistration> allBuiltInBuildCacheServices) {
         this.instantiator = instantiator;
         this.registrations = Sets.newHashSet(allBuiltInBuildCacheServices);
-
-        // By default the local cache is a directory cache
-        this.local = createLocalCacheConfiguration(instantiator, DirectoryBuildCache.class, registrations);
+        this.local = createLocalCacheConfiguration(instantiator, registrations);
     }
 
     @Override
-    public BuildCache getLocal() {
+    public DirectoryBuildCache getLocal() {
         return local;
     }
 
     @Override
-    public <T extends BuildCache> T local(Class<T> type) {
+    public <T extends DirectoryBuildCache> T local(Class<T> type) {
         return local(type, Actions.doNothing());
     }
 
     @Override
-    public <T extends BuildCache> T local(Class<T> type, Action<? super T> configuration) {
+    public <T extends DirectoryBuildCache> T local(Class<T> type, Action<? super T> configuration) {
         if (!type.equals(DirectoryBuildCache.class)) {
```

Comment:
How about we start nagging already and adjust the test when we bump master to `6.0`?

Reply:
I'd rather handle these separately. There's already the PR that does this scheduled for 6.0.